policy/modules/contrib/qgs.te

```diff
@@ -0,0 +1,74 @@
+policy_module(qgs, 1.0.0)
+
+########################################
+#
+# Declarations
+#
+
+type qgs_t;
+type qgs_exec_t;
+init_daemon_domain(qgs_t, qgs_exec_t)
+
+permissive qgs_t;
+
+type qgs_var_lib_t;
+files_type(qgs_var_lib_t)
+
+type qgs_var_run_t;
+files_type(qgs_var_run_t)
+files_pid_file(qgs_var_run_t)
+files_create_var_run_dirs(qgs_t)
+
+# Common config file for potentially many SGX
+# related apps doing SGX quoting
+type sgx_qcnl_etc_t;
+files_type(sgx_qcnl_etc_t)
+
+# Config file exclusively for SGX
+type qgs_etc_t;
+files_type(qgs_etc_t)
+
+########################################
+#
+# qgs local policy
+#
+allow qgs_t self:fifo_file rw_fifo_file_perms;
+allow qgs_t self:unix_stream_socket create_stream_socket_perms;
+
+# /var/lib/qgs is the $HOME for 'qgs' and it caches some
+# data under subdirs
+manage_dirs_pattern(qgs_t, qgs_var_lib_t, qgs_var_lib_t)
+manage_files_pattern(qgs_t, qgs_var_lib_t, qgs_var_lib_t)
+manage_lnk_files_pattern(qgs_t, qgs_var_lib_t, qgs_var_lib_t)
+files_var_lib_filetrans(qgs_t, qgs_var_lib_t, { dir file lnk_file })
+
+# /run/tdx/qgs is where 'qgs' creates UNIX socket
+manage_dirs_pattern(qgs_t, qgs_var_run_t, qgs_var_run_t)
+manage_files_pattern(qgs_t, qgs_var_run_t, qgs_var_run_t)
+manage_sock_files_pattern(qgs_t, qgs_var_run_t, qgs_var_run_t)
+files_pid_filetrans(qgs_t, qgs_var_run_t, { dir file sock_file })
+
+domain_use_interactive_fds(qgs_t)
+
+files_read_etc_files(qgs_t)
+
+# It dynmically loads sgx qcnl which reads config
+# from /etc/sgx_default_qcnl.conf
+files_config_file(sgx_qcnl_etc_t)
+read_files_pattern(qgs_t, sgx_qcnl_etc_t, sgx_qcnl_etc_t)
+
+# To read /etc/qgs.conf for its configuration
+files_config_file(qgs_etc_t)
+read_files_pattern(qgs_t, qgs_etc_t, qgs_etc_t)
+
+logging_send_syslog_msg(qgs_t)
```

Comment:
calls to logging, miscfiles, sysnetwork interfaces should be in an optional block

Reply:
I am wrapping these in optional now.